packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -105,6 +97,38 @@ class CurlyComponentManager {
     return bucket;
   }
 
+  ensureCompilable(definition, bucket, env) {
+    if (definition.template) {
+      return definition;
+    }
+
+    let { component } = bucket;
+    let TemplateFactory = component.layout;
+    let template;
+    let owner = getOwner(component);
+
+    // If the user has supplied their own layout property we must
+    // create an instance of the wrapper class with the env to go back into
+    // the runtime compiler phase :(
+    if (TemplateFactory) {
+      template = new TemplateFactory(env);
```

Comment:
when we create the factory, if this component is rendered again, we won't find the definition and recreate and recompile. I was caching in lateBound() but I'm realizing we need a way to identify the template itself.